src/platforms/hosted/cmsis_dap.c

```diff
@@ -484,7 +484,7 @@ static void dap_mem_read(adiv5_access_port_s *ap, void *dest, uint32_t src, size
 		return;
 	}
 	/* Otherwise proceed blockwise */
-	const size_t blocks_per_transfer = (report_size - 4U) >> 2U;
+	const size_t blocks_per_transfer = (report_size - 1 - DAP_CMD_BLOCK_READ_HDR_LEN) >> 2U;
```

Comment:
Why the `- 1` (which should have a U on it anyway to stop the signed-unsigned conversion) here? We agree with the macros, their introduction and value, but we don't see why you're subtracting 1 here and what that's hoping to accomplish (same for the one below)

Reply:
Good question! `dbg_dap_cmd_hid` needs one byte for the HID Report ID, so actually this byte is only needed in HID mode not bulk mode.

I'll make this self-documenting and update the PR.

Reply:
@dragonmux PTAL. This makes some behaviour changes that I'm not able to test here (the maximum transfer length for Bulk type CMSIS adapters is now one byte longer.)

I am starting to wonder if it's too fiddly. Another way to do it would be `#define DAP_CMD_BLOCK_HDR_OVERHEAD 6U` and apply this offset everywhere - worst case, some transfers are a couple of bytes shorter than the hardware would have accepted. Let me know what you think :)

Reply:
We're easily able to test CMSIS-DAP v2 via our ORBTrace unit, so this is fine - if this works on v1 devices (HID), then we'll do the testing leg for this.

The LPC4370 and ORBTrace makes for a great test case for protocol changes like these :)

Reply:
Yes, I've been testing with a DAP-Link so CMSIS-DAP V1. The write length bug I've been triggering by running the changes in #1611. I haven't got a way to max out the read response length but it doesn't change for HID in this PR. 

ORBTrace looks like very nice hardware, but the next debugger I buy will probably be BMP :grin:

(I'm waiting for Adafruit to restock them, as the shipping to Australia is pretty pricey otherwise.)